src/js/common/store/index.js

```diff
@@ -1,8 +1,23 @@
-import { createStore, applyMiddleware, compose } from 'redux';
+import _ from 'lodash/fp';
+import { createStore, applyMiddleware, compose, combineReducers } from 'redux';
 import thunk from 'redux-thunk';
-import reducer from '../reducers';
+import login from '../../login/reducers/login';
+import profile from '../../user-profile/reducers/profile';
 
-export const commonStore = createStore(reducer, compose(
-  applyMiddleware(thunk),
-  window.devToolsExtension ? window.devToolsExtension() : f => f
-));
+export const commonReducer = {
+  user: combineReducers({
+    login,
+    profile
+  })
+};
+
+export default function createCommonStore(appReducer = {}) {
+  const reducer = _.assign(appReducer, commonReducer);
+
+  return createStore(combineReducers(reducer), compose(
+    applyMiddleware(thunk),
+    __BUILDTYPE__ === 'development' && window.devToolsExtension
```

Comment:
The short-hand if-statement syntax looks a bit gnarly.  A named function might read better.

This is a _very_ nitpicky comment, feel free to disregard at your discretion.